$OpenBSD: patch-lib_AnyEvent_TLS_pm,v 1.1 2021/09/17 12:26:39 bluhm Exp $

LibreSSL uses TLS 1.3.  This is not expected by AnyEvent.  Especially
an EPIPE error confuses the t/80_ssltest.t test.

Index: lib/AnyEvent/TLS.pm
--- lib/AnyEvent/TLS.pm.orig
+++ lib/AnyEvent/TLS.pm
@@ -629,6 +629,7 @@ sub new {
 
    my $op = Net::SSLeay::OP_ALL ();
 
+   $op |= Net::SSLeay::OP_NO_TLSv1_3    (); # LibreSSL EPIPE is not expected
    $op |= Net::SSLeay::OP_NO_SSLv2      () unless $arg{sslv2};
    $op |= Net::SSLeay::OP_NO_SSLv3      () if exists $arg{sslv3} && !$arg{sslv3};
    $op |= Net::SSLeay::OP_NO_TLSv1      () if exists $arg{tlsv1} && !$arg{tlsv1};
